packages/llvm.rb

```diff
@@ -285,52 +113,67 @@ def self.build
 cxx_inc=#{CREW_PREFIX}/include/c++/\${version}/\${machine}
 gnuc_lib=#{CREW_LIB_PREFIX}/gcc/\${machine}/\${version}
 clang++ -fPIC  -rtlib=compiler-rt -stdlib=libc++ -cxx-isystem \${cxx_sys} -I \${cxx_inc} -B \${gnuc_lib} -L \${gnuc_lib} \"\$@\"' > clc++"
-      case ARCH
-      when "x86_64"
-        system 'cmake',
-               "-DCURSES_INCLUDE_PATH='#{CREW_PREFIX}/include/ncursesw'",
-               "-DCMAKE_INSTALL_PREFIX=#{CREW_PREFIX}",
-               '-DCMAKE_CXX_FLAGS="-fPIC"',
-               '-DCMAKE_BUILD_TYPE=Release',
-               '-DLLVM_LIBDIR_SUFFIX=64',
-               '-DBUILD_SHARED_LIBS=ON',
-               '-DLLVM_ENABLE_RTTI=ON',
-               '-Wno-dev',
-               '..'
-               # Fix for file INSTALL cannot find ".../lib64/python2.7" error.
-               # See http://lists.llvm.org/pipermail/lldb-dev/2015-June/007633.html.
-               system "sed -i '40,43d' tools/lldb/cmake_install.cmake"
-      else # armv7l, aarch64 or i686
-        system 'cmake',
-               "-DCURSES_INCLUDE_PATH='#{CREW_PREFIX}/include/ncursesw'",
-               "-DCMAKE_INSTALL_PREFIX=#{CREW_PREFIX}",
-               '-DCMAKE_CXX_FLAGS="-fPIC"',
-               '-DCMAKE_BUILD_TYPE=Release',
-               '-DBUILD_SHARED_LIBS=ON',
-               '-DLLVM_ENABLE_RTTI=ON',
-               '-Wno-dev',
-               '..'
-      end
-      system 'make'
+        system "env PATH=#{CREW_LIB_PREFIX}/ccache/bin:#{CREW_PREFIX}/bin:/usr/bin:/bin LD=ld.lld \
+            cmake -G Ninja \
+            -DLLVM_ENABLE_LTO=Thin \
+            -DCMAKE_C_COMPILER=$(which clang) \
+            -DCMAKE_CXX_COMPILER=$(which clang++) \
+            -DCMAKE_INSTALL_PREFIX=#{CREW_PREFIX} \
+            -DLLVM_DEFAULT_TARGET_TRIPLE=#{LLVM_DEFAULT_TARGET_TRIPLE} \
+            -DCMAKE_C_COMPILER_TARGET=#{LLVM_DEFAULT_TARGET_TRIPLE} \
+            -DCMAKE_ASM_COMPILER_TARGET=#{LLVM_DEFAULT_TARGET_TRIPLE} \
+            -DLLVM_TARGETS_TO_BUILD=\'#{LLVM_TARGETS_TO_BUILD}' \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DLLVM_LIBDIR_SUFFIX='#{CREW_LIB_SUFFIX}' \
+            -DCMAKE_LINKER=$(which ld.lld) \
+            -D_CMAKE_TOOLCHAIN_PREFIX=llvm- \
+            -DLLVM_BUILD_LLVM_DYLIB=ON \
+            -DLLVM_LINK_LLVM_DYLIB=ON \
+            -DLLVM_CCACHE_BUILD=ON \
+            -DLLVM_ENABLE_FFI=ON \
+            -DLLVM_ENABLE_RTTI=ON \
+            -DCMAKE_C_FLAGS='#{@ARCH_C_LTO_FLAGS}' \
+            -DCMAKE_CXX_FLAGS='#{@ARCH_CXX_LTO_FLAGS}' \
+            -DCMAKE_EXE_LINKER_FLAGS='#{@ARCH_LDFLAGS}' \
+            -DLLVM_PARALLEL_LINK_JOBS=1 \
+            -DPYTHON_EXECUTABLE=$(which python3) \
+            -DLLVM_INSTALL_UTILS=ON \
+            -DLLVM_BINUTILS_INCDIR='../binutils/include' \
```

Comment:
Could you reference the files from the `binutils` package instead?

Reply:
That actually should probably work... The documentation says it is looking for plugin-api.h as per https://llvm.org/docs/GoldPlugin.html which is in the binutils package.